### ISSUE TITLE
Rule Change: modify no-use-before-define to allow references to arrow functions on classes without erroring

### ISSUE THREAD

**Issue by Buttars** [eslint/eslint#17454]
### What rule do you want to change?

no-use-before-define

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
class ExampleClass {
  variable = this.exampleFunction();
  constructor() {}

  // This should be allowable, but it isn't. 
  exampleFunction = () => {}
  // This should also be allowable
  exampleImplicitReturnFunction = () => "ipsum lorem"
}
```

### What does the rule currently do for this code?

The variable option for this rule gives no option to differentiate between arrow functions on classes. I typically use arrow functions everywhere and would like to do so on classes as well without violating the no-use-before-define variable rule.

### What will the rule do after it's changed?

Have an option to exclude arrow functions on classes when checking for variables used before being defined.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Please let me know if this is a good first issue. I'm unfamiliar with the inner workings of eslint but willing to contribute if I can.

**Comment by Rec0iL99** [comment]
Hi @Buttars, thanks for the issue. I'm not able to understand the exact problem you face in the example you provided. Could you please include a [playground link](https://eslint.org/play) demonstrating the lint error you face?

**Comment by Buttars** [comment]
> Hi @Buttars, thanks for the issue. I'm not able to understand the exact problem you face in the example you provided. Could you please include a [playground link](https://eslint.org/play) demonstrating the lint error you face?

This was an issue I observed on a co-workers environment. I tried to replicate the issue on my machine and was unable to get it to error.

I'm going to close for now and reopen if it comes back up.
